rasterio/features.py

```diff
@@ -483,3 +490,150 @@ def is_valid_geom(geom):
                 return False  # short-circuit and fail early
 
     return True
+
+
+def dataset_features(
+        src,
+        bidx,
+        sampling=1,
+        band=True,
+        as_mask=False,
+        with_nodata=False,
+        geographic=True,
+        precision=-1):
+    """Yield GeoJSON features for the dataset
+
+    The geometries are polygons bounding contiguous regions of the same raster value.
+
+    Parameters
+    ----------
+    src: Rasterio Dataset
+
+    bidx: int
+        band index
+
+    sampling: int (DEFAULT: 1)
+        Inverse of the sampling fraction; a value of 10 decimates
+
+    band: boolean (DEFAULT: True)
+        extract features from a band (True) or a mask (False)
+
+    as_mask: boolean (DEFAULT: False)
+        Interpret band as a mask and output only one class of valid data shapes?
+
+    with_nodata: boolean (DEFAULT: False)
+        Include nodata regions?
+
+    geographic: str (DEFAULT: True)
+        Output shapes in EPSG:4326? Otherwise use the native CRS.
+
+    precision: int (DEFAULT: -1)
+        Decimal precision of coordinates. -1 for full float precision output
```

Comment:
@sgillies I chose to make everything but `src` and `bidx` a kwarg with defaults matching the CLI defaults. ok?

Reply:
Perfect.